components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2627,6 +2628,42 @@ where
         self.fsm.hibernate_state.count_vote(from.get_id());
     }
 
+    fn on_availability_request(&mut self, from: &metapb::Peer, msg: &ExtraMessage) {
+        if !self.fsm.peer.is_leader() {
+            return;
+        }
+        if !msg.wait_data {
+            self.fsm
+                .peer
+                .wait_data_peers
+                .retain(|id| *id != from.get_id());
+            debug!(
+                "receive peer ready info";
+                "peer_id" => self.fsm.peer.peer.get_id(),
+            );
+            return;
+        }
+        self.register_check_peers_availability_tick();
```

Comment:
If `wait_data_peers` is empty, no need to register it again.

Reply:
I think the previous if judgment guarantees this：
- If `msg.wait_data` is true, the peer should still in `wait_data_peers`, so it's not empty.
- If `msg.wait_data` is false and already remove from `wait_data_peers` through `push`, it'll return directly, so won't register it again.